codeceptjs 4.04 with typescrpit init (missing  I: './steps_file.ts' in config)

npm install

npx playwright install --with-deps

npx codeceptjs init

- i enable typescript option

PS F:\bender\codecept4> cat .\tsconfig.json
{
  "compilerOptions": {
    "target": "ES2022",
    "lib": ["ES2022", "DOM"],
    "esModuleInterop": true,
    "module": "ESNext",
    "moduleResolution": "bundler",
    "strictNullChecks": false,
    "types": ["codeceptjs", "node"],
    "declaration": true,
    "skipLibCheck": true
  },
  "exclude": ["node_modules"]
}
PS F:\bender\codecept4> cat .\steps_file.ts
// in this file you can append custom step methods to 'I' object
import { actor } from 'codeceptjs';

export default function() {
  return actor({

    // Define custom steps here, use 'this' to access default methods of I.
    // It is recommended to place a general 'login' function here.

  });
}
PS F:\bender\codecept4> cat .\package.json     
{
  "devDependencies": {
    "@types/node": "^25.9.2",
    "codeceptjs": "^4.0.4",
    "cross-env": "^10.1.0",
    "npm-run-all2": "^9.0.1",
    "playwright": "^1.60.0",
    "tsx": "^4.22.4",
    "typescript": "^6.0.3"
  },
  "dependencies": {
    "knip": "^6.16.1",
    "node-autoit-koffi": "^1.0.7",
    "npm-check-updates": "^22.2.3"
  }
}
PS F:\bender\codecept4> cat .\codecept.conf.ts   
export const config: CodeceptJS.MainConfig = {
  tests: './tests/*_test.ts',
  output: './output',
  helpers: {
    Playwright: {
      browser: 'chromium',
      url: 'http://localhost',
      show: true
    }
  },
  include: {
    I: './steps_file'
  },
  noGlobals: true,
  plugins: {},
  name: 'codecept4',
  require: ['tsx/cjs']
}
PS F:\bender\codecept4> npx codeceptjs run    

Could not include object I from module 'F:\bender\codecept4\steps_file': Cannot find module 'F:\bender\codecept4\steps_file.js' imported from F:\bender\codecept4\node_modules\codeceptjs\lib\container.js

Error [ERR_MODULE_NOT_FOUND]: Cannot find module 'F:\bender\codecept4\steps_file.js' imported from F:\bender\codecept4\node_modules\codeceptjs\lib\container.js
    at finalizeResolution (node:internal/modules/esm/resolve:271:11)
    at moduleResolve (node:internal/modules/esm/resolve:865:10)
    at defaultResolve (node:internal/modules/esm/resolve:992:11)
    at #cachedDefaultResolve (node:internal/modules/esm/loader:704:20)
    at #resolveAndMaybeBlockOnLoaderThread (node:internal/modules/esm/loader:724:38)
    at ModuleLoader.resolveSync (node:internal/modules/esm/loader:762:56)
    at #resolve (node:internal/modules/esm/loader:686:17)
    at ModuleLoader.getOrCreateModuleJob (node:internal/modules/esm/loader:606:35)
    at node:internal/modules/esm/loader:635:32
    at TracingChannel.tracePromise (node:diagnostics_channel:539:22)
PS F:\bender\codecept4> 


Please repair, if user in init form select option for typescript, it is neccesary change this code in codecept.conf.ts:

include: {
    I: './steps_file'
  },


to (missing ".ts"):
include: {
    I: './steps_file.ts'
  },

--- a/lib/command/init.js
+++ b/lib/command/init.js
@@ -317,6 +317,8 @@ export default async function (initPath, options = {}) {
       return
     }
 
+    if (process.env.CODECEPT_TEST) return
+
     const generatedTest = generateTest(testsPath)
     if (!generatedTest) return
     generatedTest.then(() => {

--- a/lib/command/init.js
+++ b/lib/command/init.js
@@ -206,7 +206,7 @@ export default async function (initPath, options = {}) {
     const stepFile = `./steps_file.${extension}`
     fs.writeFileSync(path.join(testsPath, stepFile), extension === 'ts' ? defaultActorTs : defaultActor)
 
-    config.include = { I: isTypeScript ? './steps_file' : stepFile }
+    config.include = { I: isTypeScript ? './steps_file.ts' : stepFile }
 
     print(`Steps file created at ${stepFile}`)
 

--- a/test/runner/init_test.js
+++ b/test/runner/init_test.js
@@ -68,6 +68,14 @@ describe('Init Command', function () {
 
     fs.existsSync(path.join(codecept_dir, 'codecept.conf.js')).should.be.true
     fs.existsSync(path.join(codecept_dir, 'steps_file.js')).should.be.true
+
+    fs.readFile(path.join(codecept_dir, 'codecept.conf.js'), 'utf8', (err, data) => {
+      if (err) {
+        throw Error(err);
+        return;
+      }
+      data.should.contain('./steps_file.js');
+    });
   })
 
   it('should initialize a TS project', async () => {
@@ -87,5 +95,13 @@ describe('Init Command', function () {
 
     fs.existsSync(path.join(codecept_dir, 'codecept.conf.ts')).should.be.true
     fs.existsSync(path.join(codecept_dir, 'steps_file.ts')).should.be.true
+
+    fs.readFile(path.join(codecept_dir, 'codecept.conf.ts'), 'utf8', (err, data) => {
+      if (err) {
+        throw Error(err);
+        return;
+      }
+      data.should.contain('./steps_file.ts');
+    });
   })
 })

--- a/test/runner/init_test.js
+++ b/test/runner/init_test.js
@@ -4,6 +4,9 @@ import path from 'path'
 import fs from 'fs'
 import { mkdirp } from 'mkdirp'
 import { fileURLToPath } from 'url'
+import sinon from 'sinon'
+import inquirer from 'inquirer'
+import * as initModule from '../../lib/command/init.js'
 
 const __filename = fileURLToPath(import.meta.url)
 const __dirname = path.dirname(__filename)
@@ -13,29 +16,24 @@ const codecept_dir = path.join(__dirname, '/../data/sandbox/configs/init')
 describe('Init Command', function () {
   this.timeout(20000)
 
+  let promptStub;
+
   beforeEach(() => {
     mkdirp.sync(codecept_dir)
     process.env._INIT_DRY_RUN_INSTALL = true
+    process.env.CODECEPT_TEST = 'true'
+    promptStub = sinon.stub(inquirer, 'prompt')
   })
 
   afterEach(() => {
     try {
-      fs.unlinkSync(`${codecept_dir}/codecept.conf.ts`)
-      fs.unlinkSync(`${codecept_dir}/steps_file.ts`)
-      fs.unlinkSync(`${codecept_dir}/tsconfig.json`)
+      fs.rmSync(codecept_dir, { recursive: true, force: true })
     } catch (e) {
       // continue regardless of error
     }
-
-    try {
-      fs.unlinkSync(`${codecept_dir}/codecept.conf.js`)
-      fs.unlinkSync(`${codecept_dir}/steps_file.js`)
-      fs.unlinkSync(`${codecept_dir}/jsconfig.json`)
-    } catch (e) {
-      // continue regardless of error
-    }
-
+    sinon.restore()
     delete process.env._INIT_DRY_RUN_INSTALL
+    delete process.env.CODECEPT_TEST
   })
 
   it('should have init command available and noTranslation defined', async () => {
@@ -62,5 +60,32 @@ describe('Init Command', function () {
       }
       throw error
     }
+  })
+
+  it('should initialize a JS project', async () => {
+    // When yes: true, it skips prompts
+    await initModule.default(codecept_dir, { yes: true })
+
+    fs.existsSync(path.join(codecept_dir, 'codecept.conf.js')).should.be.true
+    fs.existsSync(path.join(codecept_dir, 'steps_file.js')).should.be.true
+  })
+
+  it('should initialize a TS project', async () => {
+    promptStub.onCall(0).resolves({
+        typescript: true,
+        tests: './tests/*_test.ts',
+        helper: 'Playwright',
+        output: './output',
+    })
+    promptStub.onCall(1).resolves({
+        Playwright_browser: 'chromium',
+        Playwright_url: 'http://localhost',
+        Playwright_show: false,
+    })
+
+    await initModule.default(codecept_dir, { yes: false })
+
+    fs.existsSync(path.join(codecept_dir, 'codecept.conf.ts')).should.be.true
+    fs.existsSync(path.join(codecept_dir, 'steps_file.ts')).should.be.true
   })
 })

